project1-sps/script.js

```diff
@@ -0,0 +1,227 @@
+var roll3Number = function () {
+  // generate decimal 0 <= x < 3
+  var randomDecimal = Math.random() * 3;
+
+  // generate integer 0 <= x <= 2
+  var randomInteger = Math.floor(randomDecimal);
+
+  var chosenNumber = randomInteger + 1;
+  return chosenNumber;
+};
+
+let userName = "";
+let gameState = "start";
+let totalGames = 0;
+let totalWin = 0;
+let totalLose = 0;
+let totalDraw = 0;
+let currentMode = 0;
+
+var main = function (input) {
+  let message;
+
+  // ask for name
+  if (gameState == "start") {
+    gameState = "username";
+    message = "Hi! Welcome to Scissors-Paper-Stone. Please key in your name.";
+    return message;
+  }
+
+  // ask for mode
+  if (gameState == "username") {
+    userName = input;
+    gameState = "mode";
+    message =
+      `Nice to meet you, ${userName}. Please select which mode (1, 2, 3 or 4) will you be playing.` +
+      "<br>" +
+      "<br>" +
+      "1: Manual Normal mode" +
+      "<br>" +
+      "2: Manual Reverse mode" +
+      "<br>" +
+      "3: AI Normal mode" +
+      "<br>" +
+      "4: AI Reverse mode";
+    return message;
+  }

```

Comment:
this is fine but strange that you're using template literals on line 35 but then using normal string concatenation with `+` for the other parts of the string. consider making it consistent